matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/query/EventEntityQueries.kt

```diff
@@ -25,9 +25,12 @@ import io.realm.kotlin.where
 
 internal fun EventEntity.copyToRealmOrIgnore(realm: Realm): EventEntity {
     return realm.where<EventEntity>()
-        .equalTo(EventEntityFields.EVENT_ID, eventId)
-        .equalTo(EventEntityFields.ROOM_ID, roomId)
-        .findFirst() ?: realm.copyToRealm(this)
+            .equalTo(EventEntityFields.EVENT_ID, eventId)
+            .equalTo(EventEntityFields.ROOM_ID, roomId)
+            .findFirst()
+            // If Event already exist, ensure state is up to date
+            ?.also { it.sendState = sendState }
+            ?: realm.copyToRealm(this)
```

Comment:
Actually there is maybe a pb here: synced event content will not replace what we had for the local echo anymore. Synced event may contain other field (in unsigned data?)